libobs/obs-encoder.c

```diff
@@ -2046,6 +2046,13 @@ uint32_t obs_encoder_get_roi_increment(const obs_encoder_t *encoder)
 	return encoder->roi_increment;
 }
 
+obs_encoder_group_t *obs_encoder_get_group(obs_encoder_t *encoder)
+{
+	return obs_encoder_valid(encoder, "obs_encoder_get_group")
+		       ? encoder->encoder_group
+		       : NULL;
+}
```

Comment:
given the intricacies around the lifetime of groups I'd suggest going with a design that doesn't expose the pointer to code that doesn't own the group, e.g.
```c++
bool obs_encoder_is_grouped_with(obs_encoder_t *encoder, obs_encoder_t *other_encoder)
```

or maybe add a comment to the function along the lines of
> this pointer is only safe to use before encoders have been started or while encoders are active; the pointer might get invalidated as soon as encoders become inactive